### PR TITLE
feat: cloudevent behavior update

### PR DESCRIPTION
**Pull request by bshaffer**
Addresses #49 

We need to verify this passes the conformance tests before merging!

### CODE REVIEW

File: src/CloudEventFunctionWrapper.php [hunk 1]
```diff
@@ -20,35 +20,230 @@
 use GuzzleHttp\Psr7\Response;
 use Psr\Http\Message\ResponseInterface;
 use Psr\Http\Message\ServerRequestInterface;
+use ReflectionFunctionAbstract;
+use LogicException;
 use RuntimeException;
 
 class CloudEventFunctionWrapper extends FunctionWrapper
 {
+    private const TYPE_LEGACY = 1;
```
Comment (grant) [inline src/CloudEventFunctionWrapper.php:29]: I'd perfer if the logic for handling legacy events were part of a separate file.

Legacy isn't a type of a HTTP protocol binding for CloudEvents

https://github.com/cloudevents/spec/blob/master/http-protocol-binding.md

**Review by grant — APPROVED** [review]
Looks fine from what I reviewed.

I would prefer if the mapping logic was separate from the CloudEventFunctionWrapper handler.

That would allow for the CloudEvent handler to be isolated, just accept CloudEvents. A separate function transforms request from non-CloudEvents to CloudEvents before sending it to this handler.

File: src/CloudEventFunctionWrapper.php (hunk 1 above)
Comment (bshaffer) [reply src/CloudEventFunctionWrapper.php:28]: Added `LegacyEventMapper` to address this. PTAL!

Comment (bshaffer) [reply src/CloudEventFunctionWrapper.php:28]: The `CloudEventFunctionWrapper` still detects the event type and converts the request to a `CloudEvent`. As we have defined the `FunctionsWrapper` classes to take a `ServerRequestInterface` in `execute`, it makes sense that converting from a Request to a CloudEvent would happen in the wrapper.

File: src/CloudEventFunctionWrapper.php [hunk 2]
```diff
@@ -20,35 +20,100 @@
...
-        // Get Headers
-        $headers = $request->getHeaders();
-        $cloudeventContent = [];
-        $validKeys = ['id', 'source', 'specversion', 'type', 'datacontenttype', 'dataschema', 'subject', 'time'];
-        foreach ($validKeys as $key) {
+
+        return $jsonData;
+    }
+
+    private function fromCloudEvent(
```
Comment (grayside) [inline src/CloudEventFunctionWrapper.php:98]: nit: shouldn't this be `fromHttpRequest()`?

Comment (bshaffer) [reply src/CloudEventFunctionWrapper.php]: I agree! Updated to `fromRequest`

**Review by grayside — APPROVED** [review]

**Pull request merged**